packages/react/src/components/Pagination/Pagination.tsx

```diff
@@ -1,112 +1,51 @@
-import { forwardRef } from 'react';
-import type { HTMLAttributes } from 'react';
+import { Slot } from '@radix-ui/react-slot';
+import cl from 'clsx/lite';
+import { createContext, forwardRef, useState } from 'react';
+import type { Dispatch, HTMLAttributes, SetStateAction } from 'react';
 
-import { PaginationButton } from './PaginationButton';
-import { PaginationEllipsis } from './PaginationEllipsis';
-import { PaginationItem } from './PaginationItem';
-import { PaginationList } from './PaginationList';
-import { PaginationNext, PaginationPrevious } from './PaginationNextPrev';
-import { PaginationRoot } from './PaginationRoot';
-import { usePagination } from './usePagination';
+export const PaginationContext = createContext({
+  size: 'md' as NonNullable<PaginationProps['size']>,
+});
```

Comment:
If we are using a hook for pagination, why do we need a context? Could we just add the size to the button props that is already being spread? Or simply set `data-size="*"` since we are omitting the type